### PR TITLE
feat(class): Class Details Register Modal Flow

### PR DESCRIPTION
**Pull request by MatoPlus**
### Notion ticket link

[Class Details Register Modal Flow](https://www.notion.so/uwblueprintexecs/Class-Details-Register-Modal-Flow-6d7153d995c447898064a3ce05316c41)

https://user-images.githubusercontent.com/44826218/134452359-aaf75acd-0185-4304-9ce3-22f2d6b5b3df.mp4

https://user-images.githubusercontent.com/44826218/134452381-3926ea9b-b94d-4a02-a7a7-161e142c81f2.mp4

https://user-images.githubusercontent.com/44826218/134452392-9ff8bcb8-41a8-4b32-badc-8fa57cf40a9b.mp4

### Implementation description

- Used Chakra UI modal (similar to the language modal) to implement class details registration workflow
- The routing of the button depends on whether the current user has a session (routes to `/parent-enrollment` or `/login`)
- Add modal for ineligible class prompts
- Made and used several util methods for date and time range formatting to match look of Figma

### Steps to test

1. Frontend implementions satisfies designer's intentions (see video attachments)

1. Create several classes and programs with varying time range, dates
2. Click try to register by clicking on classes
3. Ensure that dates, times, class details shows up as intended and matches Figma designs

### What should reviewers focus on?

- How's the look and feel of the modals?
- Are the util methods general and robust enough?
- Are the modal used properly in the code implemention?

### Checklist

-   [X] My PR name is descriptive and in imperative tense
-   [X] I have run the linter
-   [X] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by MatoPlus** [comment]
You may have noticed from the demo videos that redirecting to a new route does not scroll the page to the top, this has been fixed

https://user-images.githubusercontent.com/44826218/134607458-efef8157-8303-4909-af3c-6a7e9752d02d.mp4

:

File: src/components/ClassInfoModal.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                    <Button
+                        bg={"#0C53A0"}
+                        color={"white"}
+                        mx={"auto"}
+                        my={2}
+                        onClick={onRegister}
+                        fontWeight={"200"}
+                        _hover={{
+                            textDecoration: "none",
+                            bg: "#2C6AAD",
```
Comment (MatoPlus) [inline src/components/ClassInfoModal.tsx]: To self: we can use the colourTheme helper

**Review by feng-eric — APPROVED** [review]
LGTM as long as you have tested it!

**Review by bekahma — COMMENTED** [review]
[design] everything looks good to me! the modal flow is good - one thing i noticed is the date is displayed differently in the modal compared to the main page (**July 22 to September 29, 2021** vs. **Sep 22 2021 to Sep 29 2021**)

**Comment by MatoPlus** [comment]
> [design] everything looks good to me! the modal flow is good - one thing i noticed is the date is displayed differently in the modal compared to the main page (**July 22 to September 29, 2021** vs. **Sep 22 2021 to Sep 29 2021**)

Good catch! Straight up an oversight on my part. Resolved.

image

**Review by bekahma — APPROVED** [review]
thanks for fixing the date format @MatoPlus 😄

**Review by gregmaxin — APPROVED** [review]

**Pull request merged**